agent/agent.go

```diff
@@ -537,6 +537,9 @@ func (a *Agent) Start() error {
 	a.httpConnLimiter.SetConfig(connlimit.Config{
 		MaxConnsPerClientIP: a.config.HTTPMaxConnsPerClient,
 	})
+	if err := lib.CheckLimitsFromMaxConnsPerClient(a.config.HTTPMaxConnsPerClient); err != nil {
+		return err
```

Comment:
If I am reading the code correctly this value is already validated by `Builder.Validate()` which is called from `cmd.readConfig()`->`Builder.BuildAndValidate()`->`Builder.Validate()` before it calls`Agent.Start()`.

If that is correct then I do not think we need this call here. The value would already have been validated by the the call to Builder.Validate().